Makefile

```diff
@@ -0,0 +1,25 @@
+.PHONY: all clean plugin docker-install
+.SUFFIXES:
+
+version := $(shell grep '^    <version>.*</version>' pom.xml | cut -d '>' -f 2 | cut -d '<' -f 1)
+plugin_name := gocd-riemann-notifier-$(version).jar
+
+plugin_path := target/$(plugin_name)
+install_path := docker/gocd-server/plugins/external/$(plugin_name)
+
+all: plugin
+
+clean:
+	rm -rf target
+
+$(plugin_path): pom.xml $(shell find src -type f)
```

Comment:
Did globs not work?
```suggestion
$(plugin_path): pom.xml src/**/*.java
```

Reply:
I get this error if I do this:
```
make: *** No rule to make target `src/main/*.java', needed by `target/gocd-riemann-notifier-0.8.jar'.
```

Reply:
Ah hmm - it turns out this works in the other project I looked at purely by chance. 🤦‍♂️ Globs _do_ work in recipe dependencies, but it turns out `**` is _not_ a recursive wildcard in `make` - it's literally "match any string, then match any string next to it". This works if you specify the same number of wildcards as you have directory levels (or explicitly specify them):
```
% tree src
src
└── foo
    ├── bar
    │   ├── a.txt
    │   └── b.txt
    └── baz
        └── c.txt

% cat Makefile 
example.zip: src/*/*/*
        zip $@ $^

% make
zip example.zip src/foo/bar/a.txt src/foo/bar/b.txt src/foo/baz/c.txt
  adding: src/foo/bar/a.txt (stored 0%)
  adding: src/foo/bar/b.txt (stored 0%)
  adding: src/foo/baz/c.txt (stored 0%)
```